1. Basic Web Development/frontend/src/main.ts

```diff
@@ -0,0 +1,419 @@
+
+    removeChildren(permissionsList);
+
+    this.PERMISSIONS.forEach((element: Permission) => {
+        const newElement = document.createElement("div");
+        newElement.classList.add("grid-element");
+
+        const newLabel = document.createElement("label");
+        newLabel.setAttribute('for', element.id);
+        newLabel.innerHTML = element.text;
+
+        const newInput = document.createElement("input");
+        newInput.setAttribute('id', element.id);
+        newInput.setAttribute('name', element.type);
+        newInput.setAttribute('type', 'checkbox');
+
+        if (element.type == "permissionNss") {
+            newInput.click();
+            newInput.disabled = true;
+        }
+
+        const newButtonDelete = document.createElement("button");
+        newButtonDelete.setAttribute('type', 'button');
+        newButtonDelete.textContent = 'X';
+        newButtonDelete.setAttribute('name', element.id);
+        newButtonDelete.classList.add('letter-btn');
+        newButtonDelete.classList.add('delete-btn');
+
+        const newButtonEdit = document.createElement("button");
+        newButtonEdit.setAttribute('type', 'button');
+        newButtonEdit.textContent = 'E';
+        newButtonEdit.setAttribute('name', element.id);
+        newButtonEdit.classList.add('letter-btn');
+        newButtonEdit.classList.add('edit-btn');
+
+        newElement.appendChild(newButtonEdit);
+        newElement.appendChild(newButtonDelete);
+        newElement.appendChild(newLabel);
+        newElement.appendChild(newInput);
+
+        permissionsList.appendChild(newElement);
+    });
+
+    setCounter();
+
+    this.BOXES = getBoxes();
+}
+
+function setCounter() {
+    const counterDisplay = document.getElementById("counter-display");
+    removeChildren(counterDisplay);
+
+    this.COUNTER = document.querySelectorAll('input[type="checkbox"]:checked').length;
+
+    const newText = document.createElement('span');
+    newText.textContent = `${this.COUNTER} permissions`;
+    counterDisplay.appendChild(newText);
+}
+
+function handleRadios() {
+    document.addEventListener('change', event => {
+        if ((event.target as HTMLInputElement).name == "preset") {
+            switch ((event.target as HTMLInputElement).value) {
+                case "All":
+                    this.changeBoxesState(this.BOXES.pmsBoxes, true);
+                    this.changeBoxesState(this.BOXES.allBoxes, true);
+                    break;
+
+                case "Permissive":
+                    this.changeBoxesState(this.BOXES.pmsBoxes, true);
+                    this.changeBoxesState(this.BOXES.allBoxes, false);
+                    break;
+
+                case "Necessary":
+                    this.changeBoxesState(this.BOXES.pmsBoxes, false);
+                    this.changeBoxesState(this.BOXES.allBoxes, false);
+                    break;
+
+                default:
+                    break;
```

Comment:
🔥 `PRO TIP:` If you move files around in the folder structure on the same PR. github won't be able to figure out the `main.ts` in `root/frontend/src/` was moved here from `root/`. If this happens all comments on the initial file will be marked as outdated but that doesn't mean they're solved. Please make sure all the comments @CalinaCristian left were addressed in a way or another.

Previously here you were logging an error and Cristi mentioned you should have some mechanism to handle this.

Reply:
I removed that line because it's no longer possible for that error to appear. The only way to trigger it was if I made a typo while writing an id in the code.